PassportJS (GitHub), MongoDB, GraphQL + User and Connection Types

the-vampiire opened the pull request:
Completed through Query resolvers for all Types

Pause for unit testing then continue to Mutations

belcurv commented:
Looks good. A couple questions / comments:

In `app.js`, what happens when `app.listen()` and `mongoose.connect()` throw errors? Who catches those errors?

Something to consider: create and export collection "constants" from `/db/index`; use them in `/db/connection` & `/db/user` in place of the string names. This helps avoid spelling errors and you gain intellisense / code autocompletion.

```js
const { User } = require('./user');
const { Connection } = require('./connection');

// Expose a COLLECTIONS object.
// This just maps string name values of the collections we're interacting with.
const COLLECTIONS = {
  USER       : 'user',
  CONNECTION : 'connection'
};

module.exports = {
  COLLECTIONS,
  User,
  Connection,
};
```

Then, for ex., in: `db/user.js`:

```js
const mongoose = require('mongoose');
const { COLLECTIONS } = require('./index');

/* ... */

UserSchema.methods.ownedConnections = function ownedConnections() {
  return this.model(COLLECTIONS.CONNECTION).find({ ownerID: this.id });
};
```

Finally, there's some very minor spelling/grammatical errors in comments - I don't care, and most of those comments won't survive long (they're notes to us non-graphql wizards).

ziggysauce commented:
@Bigghead yeah, vamp just helped me catch it! Thanks!

the-vampiire commented:
@belcurv great suggestion man. one note is that it must be a standalone file. exported from index would cause a circular dependency issue. as for handling the connect and listen errors i would prefer to have the server crash - in either scenario the app is non-functional and should be taken offline.

### Updates to PR
- @Bigghead @ziggysauce: fixed `passportConfig.js` import destructuring and parameter name
- @belcurv: added Collection Names file to prevent potential string checking errors
- corrected the `passportConfig.js` serialize and deserialize methods so that only the `_oid` of the user is serialized. on deserializing the user is found and returned by id
